lib/logdna/client.rb

```diff
@@ -125,13 +120,16 @@ def flush
           send_request
         end
         @lock.unlock
+        Thread.current.exit
       else
         schedule_flush
       end
     end
 
     def exitout
-      flush
+      if @buffer.any? || @side_messages.any?
+        Thread.new { flush }
```

Comment:
Would we want this to block, since if the process ends before the thread does, the thread will (?) get aborted?

Reply:
If we wanted it to block we would need to call `Thread.join`, otherwise I believe the thread will be forcefully terminated.

I actually think we don't need a new thread on `exitout`.  it should just be apart of the main process.

Reply:
Changed this so that the new thread calls `.join` so that we make it blocking.  Easier than putting conditional logic in the `flush` method to only call `exit` on any thread other than main thread